test/python/topology/test1.py

```diff
@@ -110,6 +110,37 @@ def test_TopologySourceAndSinkCallable(self):
       topo = Topology("test_TopologySourceAndSinkCallable")
       hw = topo.source(test_functions.SourceTuplesAppendIndex(["a", "b", "c", "d"]))
       hw.sink(test_functions.CheckTuples(["a0", "b1", "c2", "d3"]))
+    
```

Comment:
I don't see a test which actually validates that the data path was parallelized. For example, if you call `hw.parallel(4)`, there should be a test which asserts that the number of parallel channels == 4.